### PR TITLE
#165085404 Design Top and Side Nav Bars

### PR DESCRIPTION
**Pull request by shaolinmkz**
#### What does this PR do?
Having the Top and Side Nav and all child components designed and tested

#### Description of Task to be completed?
Have the following completed:

1. Design the Top nav component
2. Design the Side nav component
3. Make the Top nav responsive for both mobile and desktop
4. Make the Side nav responsive for both mobile and desktop
5. Design the hamburger menu functionality
6. Make menu dropdown responsive to hamburger click
7. Write unit tests for all components

#### How should this be manually tested?
1. Clone this branch.
2. Run `cd artemis-ah-frontend`
3. Run `npm install` to install the dependencies.
4. Run `npm test` to run the tests.

#### What are the relevant pivotal tracker stories?
[#165085404](https://pivotaltracker.com/story/show/165085404)

#### Screenshots (if appropriate)
Screenshot 2019-04-07 at 1 50 57 AM
Screenshot 2019-04-07 at 1 52 39 AM
Screenshot 2019-04-07 at 1 53 26 AM
Screenshot 2019-04-07 at 1 55 21 AM
Screenshot 2019-04-07 at 1 56 23 AM

### CODE REVIEW

File: src/components/TopNav.jsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
+import React, { Component } from 'react';
+import { NavLink } from 'react-router-dom';
+import { createArticleIcon, notificationIcon } from '../assets/img__func/icons_svg';
+import Logo from './logo';
+import UserNavAvatar from './userNavAvartar';
+// import Hamburger from './Hamburger';
```
Comment (andela-gike) [inline src/components/TopNav.jsx]: @shaolinmkz, kindly remove the above commented code

File: src/components/TopNav.jsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+      auth: false,
+      hamburgerStateClose: 'inline-block',
+      hamburgerStateOpen: 'none',
+      reset: false,
+      menuClassStyleName: 'link-wrapper'
+    };
+    this.renderNavChildren = this.renderNavChildren.bind(this);
+    this.renderHamburgerMenu = this.renderHamburgerMenu.bind(this);
+  }
+
```
Comment (andela-gike) [inline src/components/TopNav.jsx:30]: @shaolinmkz, the use of arrow function has taken care of the need to use `.bind` in our react classes this tutorial by Cory house will comee in handy `https://medium.freecodecamp.org/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56` or `https://medium.com/@nikolalsvk/loosing-bind-this-in-react-8637ebf372cf`

File: src/views/BookmarkPage.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      <h1>This is your BOOKMARK page!!</h1>
+      <h1>This is your BOOKMARK page!!</h1>
+      <h1>This is your BOOKMARK page!!</h1>
+      <h1>This is your BOOKMARK page!!</h1>
+      <h1>This is your BOOKMARK page!!</h1>
+      <h1>This is your BOOKMARK page!!</h1>
+      <h1>This is your BOOKMARK page!!</h1>
+      <h1>This is your BOOKMARK page!!</h1>
+      <h1>This is your BOOKMARK page!!</h1>
+      <h1>This is your BOOKMARK page!!</h1>
```
Comment (andela-gike) [inline src/views/BookmarkPage.jsx]: @shaolinmkz you could have done this better by passing it as an array and using `.times` in lodash to multiple it to as much as you may want.

**Review by andela-gike — CHANGES_REQUESTED** [review]

File: src/components/TopNav.jsx (hunk 1 above)
Comment (shaolinmkz) [reply src/components/TopNav.jsx]: On it..

File: src/components/TopNav.jsx (hunk 2 above)
Comment (shaolinmkz) [reply src/components/TopNav.jsx:30]: On it..

File: src/views/BookmarkPage.jsx (hunk 3 above)
Comment (shaolinmkz) [reply src/views/BookmarkPage.jsx]: On it..

**Review by Chrismarcel — COMMENTED** [review]
Reviewed

File: src/components/Hamburger.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
```
Comment (andela-gike) [inline src/components/Hamburger.jsx]: This is yet to be destructured

File: src/components/logo.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (andela-gike) [inline src/components/logo.jsx]: @shaolinmkz same here

File: src/components/userNavAvartar.jsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (andela-gike) [inline src/components/userNavAvartar.jsx]: @shaolinmkz and here too

**Review by andela-gike — CHANGES_REQUESTED** [review]

File: src/components/Hamburger.jsx (hunk 4 above)
Comment (shaolinmkz) [reply src/components/Hamburger.jsx]: Oh!...
On it...

File: src/components/logo.jsx (hunk 5 above)
Comment (shaolinmkz) [reply src/components/logo.jsx]: DOne

File: src/components/userNavAvartar.jsx (hunk 6 above)
Comment (shaolinmkz) [reply src/components/userNavAvartar.jsx]: Done

**Review by andela-gike — APPROVED** [review]

**Pull request merged**